lib/modal.js

```diff
@@ -927,16 +927,16 @@
         if (visible || forceClose) {
           if (visible) {
           element.close();
+          element.setAttribute('aria-hidden', true);
           }
 
           this.lastClickedToggle.focus();
           vl.util.removeClass(document.body, noOverflowClass);
         } else {
           element.showModal();
+          element.setAttribute('aria-hidden', false);
           vl.util.addClass(document.body, noOverflowClass);
         }
```

Comment:
methodes trekken uit wat er in de if en de else-tak zit? 
anders zou ik het laten staan dat er meer 1 call in deze functie naar element.setAttribute('aria-hdden', ...) is

Reply:
Het is AIV code en bovendien gebuild, dus hierin probeer ik zo weinig mogelijk te wijzigen. Bij de bron heb ik PR gedaan, alternatief is werken met een nieuwe variable om maar één keer de `setAttribute` aan te roepen. Ik zal zien wat AIV zegt van de PR.

Reply:
zo weinig mogelijk wijzigen zou dan toch gewoon de !visible in visible veranderen zijn?

Reply:
Nee, de bug is dat de visible constante niet betrouwbaar is om te gebruiken. In de functie wordt de modal geopend of gesloten waardoor de initiële visible value niet meer correct is. Dit resulteert in een foutief aria-hidden attribuut.

Reply:
visible is toch geen constante? het enige dat ik zie in de oude code is dat volgens mij de !visible verkeerd is, maar kan ook zijn dat ik het niet begrijp dan.

Reply:
```
{
  key: "toggle",
  value: function toggle(element) {
    var forceClose = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
    var visible = element.hasAttribute('open');

    if (visible || forceClose) {
      if (visible) {
      element.close();
      element.setAttribute('aria-hidden', true);
      }

      this.lastClickedToggle.focus();
      vl.util.removeClass(document.body, noOverflowClass);
    } else {
      element.showModal();
      element.setAttribute('aria-hidden', false);
      vl.util.addClass(document.body, noOverflowClass);
    }
  }
}
```

Reply:
dus tenzij dat het punt is dat als forceClose true is, de aria-hidden niet mag gezet worden, kan je volgens mij nog steeds dat op 1 plaats steken?

Reply:
Nee, dat is het punt niet:

> Nee, de bug is dat de visible constante niet betrouwbaar is om te gebruiken. In de functie wordt de modal geopend of gesloten waardoor de initiële visible value niet meer correct is. Dit resulteert in een foutief aria-hidden attribuut.

Reply:
maar bij gebuilde code staat 1 lijn met element.setAttribute(...) en nu in uw review hebt ge da in 2 gesplitst? en visible is toch geen constante maar een local variable? dus die kan toch niet veranderen terwijl die functie bezig is. dus ik snap hem nog steeds niet.

Reply:
Concreet voorbeeld: toggle wordt aangeroepen en modal was niet open, visible const wordt gezet en is false. Element wordt in de toggle functie geopend via `showModal`, resultaat de modal staat open. Op het einde wordt het aria-hidden gezet met de visible constante die false is, m.a.w. aria-hidden = !visible = !false = true = verkeerd.

Reply:
Ja, de "visible" is, "was het visible voordat ik ga toggelen", dat begrijp ik. Maar in uw 2 codefragmenten hierboven staat er maar 1 keer element.setAttribute("..."), daar moet gewoon "!visible" toch "visible" worden, ipv dat het nu (in de review althans) 2 keer aangeroepen, een keer expliciet met false en een keer expliciet met true.

Reply:
Aaaah ... wacht ik zie het nu denk ik.
Die aanroep staat BUITEN de functie in de gebuilde code?

Reply:
De code wijzigingen kan je in de review zien, code in de opmerking hierboven is de originele code al dan niet gebuild.

Reply:
en dus kan je de setVisible toch gewoon in de functie zetten?

Reply:
?

Reply:
ja en dan is gewoon de !visible "omdraaien" en er "visible" van maken dus gelijk aan wat je nu gedupliceerd hebt, tenzij je de aria-hidden niet mag zetten als visible false en forceClose true

Reply:
Ik denk niet dat je het oorspronkelijk probleem begrijpt. Soms is de const visible correct, soms niet. Dit hangt van de inner logica af, daarom heb ik het `setAttribute` ontdubbeld.